chalk-slg/src/context/mod.rs

```diff
@@ -226,7 +226,9 @@ pub trait InferenceTable<C: Context>: Clone {
     ) -> Fallible<Self::UnificationResult>;
 }
 
-pub trait Substitution<C: Context>: Clone + Debug {}
+pub trait Substitution<C: Context>: Clone + Debug {
+    fn may_invalidate(&mut self, subst: &C::Substitution) -> bool;
```

Comment:
I think it's worth commenting this a bit. Something like this:

---

Tests whether this substitution -- if it were to become a confirmed answer -- would affect `subst`, which represents the aggregated results of all answers thus far. If this returns true, then the outer solving loop has a reason to continue requesting answers -- but if all strands return false, it does not, as `subst` will not be further refined. 

Be careful here: `self` and `subst` exist in distinct inference tables.

---

Actually, the latter point is making me wonder: I think we probably have to thread through the inference tables, and this is making me think I may prefer the "callback variant", in which this method doesn't exist (more below).